basic-type.bnf

```diff
@@ -28,17 +28,29 @@
 
 # Boolean
 
-<boolean> ::= true | false 
+<boolean> ::= true | false | :true | :false
 
 # Atom/Symbol
 
-<atom> ::= :<alpha_list> 
-<alpha_list> ::= <alpha> | <alpha_list><alpha_list> 
+<atom> ::= :<alpha><atom_term> | :<string> | :@
+<atom_term> ::= <alpha> | @ | <decimal> | <atom_term><atom_term>
 
 # String
 
-<string> ::= "<char_list>" 
-<char_list> ::= <alpha> | <num> | <special_characters> | <char_list><char_list> 
+<string>::="<string_term>"
+<string_term> ::= '
+    | <alpha>
+    | <decimal>
+    | <special_characters>
+    | <escape_sequence>
+    | <string_interpolation>
+    | <string_term><string_term>
+<string_interpolation> ::= #{ <expression> }
+<escape_sequence> ::= \<escape_codes>
+    | \x<decimal_digit><decimal_digit>
+    | \u<decimal_digit><decimal_digit><decimal_digit><decimal_digit>
+    | \u{<decimal>}
+<escape_codes> ::= " | ' | \ | a | b | d | e | f | n | r | s | t | v | 0 | #
```

Comment:
List escape codes bisa diliat di [getting started](https://elixir-lang.org/getting-started/sigils.html#interpolation-and-escaping-in-string-sigils)
![image](https://user-images.githubusercontent.com/5263688/96682622-45686a80-13a3-11eb-8150-0b02c21105c8.png)

atau bisa juga di [hexdocsnya elixir](https://hexdocs.pm/elixir/String.html#module-escape-characters)
![image](https://user-images.githubusercontent.com/5263688/96682743-76489f80-13a3-11eb-8103-5491293bd9e7.png)